app/controllers/attachment_references_controller.rb

```diff
@@ -0,0 +1,14 @@
+class AttachmentReferencesController < ApplicationController
+  load_resource :attachment_reference
+  before_action :authorize_resource
+
+  def show
+    redirect_to @attachment_reference.url
+  end
+
+  private
+
+  def authorize_resource
+    authorize!(:read, @attachment_reference.attachable)
```

Comment:
It improves the performance in those cases where attachments are in the contents of the records. For example, let's say this is a image in forum post, after forum post is authorised, if we authorise the attachment again, that would sacrifice some performance right ?